.devcontainer/Dockerfile

```
@@ -0,0 +1,47 @@
+    tar -xzf /tmp/graalvm.tar.gz -C ${GRAALVM_INSTALL_DIR} && \
+    # Extract the directory name from the tarball
+    GRAALVM_HOME_DIR=$(tar -tf /tmp/graalvm.tar.gz | head -n 1 | cut -d/ -f1) && \
+    # Rename the extracted directory to a consistent name
+    mv ${GRAALVM_INSTALL_DIR}/${GRAALVM_HOME_DIR} ${GRAALVM_INSTALL_DIR}/graalvm-jdk && \
+    rm /tmp/graalvm.tar.gz
+
+# A non-GraalVM JDK is required to run VS Code's Java extensions
+FROM mcr.microsoft.com/devcontainers/base:ubuntu-24.04 as jdk_installer
+ARG JDK_DOWNLOAD_URL=https://download.oracle.com/java/23/archive/jdk-23.0.2_linux-x64_bin.tar.gz
+ARG JDK_INSTALL_DIR=/opt
+RUN curl -L ${JDK_DOWNLOAD_URL} -o /tmp/jdk.tar.gz && \
```

Hadolint (2.12.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)